src/util/utils.ts

```diff
@@ -0,0 +1,39 @@
+import { TimingEvents } from "../types";
+import { observableClock } from "./observable";
+import {formatDuration} from "./text";
+
+export type FormattedDurationProps =
+    {
+        durationMs?: number,
```

Comment:
It looks like nothing ever sets this nowdays (since the `timingEvents` prop isn't optional here, it would be quite awkward to do so) so we can drop the `durationMs` prop entirely, and simplify this: `calculateAndFormatDuration` and `DurationPill` can just always take `TimingEvents` and nothing else.